docs: clarify scope of alpha (#6722)
apollographql/apollo-server: docs/source/index.mdx

>
> Apollo Server 4 introduces many across-the-board changes,  and we are working on updating our docs to reflect these changes. We have temporarily removed most of the documentation from this version and will add pieces back as we ensure they are up to date for Apollo Server 4. In the meantime, you can reference the [Apollo Server 3 docs](/apollo-server/) for more in-depth information on Apollo Server's features.
>

> > At this time, this alpha is intended primarily for community members to start to develop integrations between Apollo Server and their favorite web frameworks. Not all users will be able to upgrade their servers to this alpha; for example, this alpha cannot yet host an Apollo Gateway and does not work with `RESTDataSource`.

Reviewer: ```suggestion
> At this time, this alpha is intended primarily for community members to start to developing integrations between Apollo Server and their favorite web frameworks. Not all users can upgrade their servers to this alpha yet. For example, this alpha can't host an Apollo Gateway and doesn't work with `RESTDataSource`.
```
